crates/pathfinder/src/storage/state.rs

```diff
@@ -342,42 +309,33 @@ impl StarknetBlocksTable {
         tx: &Transaction<'_>,
         block: StarknetBlocksBlockId,
     ) -> anyhow::Result<Option<GlobalRoot>> {
-        let mut statement = match block {
-            StarknetBlocksBlockId::Number(_) => {
-                tx.prepare("SELECT root FROM starknet_blocks WHERE number = ?")
-            }
-            StarknetBlocksBlockId::Hash(_) => {
-                tx.prepare("SELECT root FROM starknet_blocks WHERE hash = ?")
-            }
-            StarknetBlocksBlockId::Latest => {
-                tx.prepare("SELECT root FROM starknet_blocks ORDER BY number DESC LIMIT 1")
-            }
-        }?;
-
-        let mut rows = match block {
-            StarknetBlocksBlockId::Number(number) => statement.query(params![number.0]),
-            StarknetBlocksBlockId::Hash(hash) => statement.query(params![hash.0.as_be_bytes()]),
-            StarknetBlocksBlockId::Latest => statement.query([]),
-        }?;
-
-        let row = rows.next().context("Iterate rows")?;
-        match row {
-            Some(row) => {
-                let root = row.get_ref_unwrap("root").as_blob().unwrap();
-                let root = StarkHash::from_be_slice(root).unwrap();
-                let root = GlobalRoot(root);
-                Ok(Some(root))
-            }
-            None => Ok(None),
+        match block {
+            StarknetBlocksBlockId::Number(number) => tx.query_row(
+                "SELECT root FROM starknet_blocks WHERE number = ?",
+                [number],
+                |row| row.get(0),
+            ),
+            StarknetBlocksBlockId::Hash(hash) => tx.query_row(
+                "SELECT root FROM starknet_blocks WHERE hash = ?",
+                [hash],
+                |row| row.get(0),
+            ),
+            StarknetBlocksBlockId::Latest => tx.query_row(
+                "SELECT root FROM starknet_blocks ORDER BY number DESC LIMIT 1",
```

Comment:
Does dropping the `LIMIT 1` improve things here? I imagine so since its a single row query?

Or no difference in effect?

Reply:
Didn't check, can't imagine how it could. This should work with `SELECT root, max(number) FROM starknet_blocks` though.